code/ship/ship.cpp

```diff
@@ -8774,7 +8778,7 @@ void ship_auto_repair_frame(int shipnum, float frametime)
 		if ( ssp->current_hits < ssp->max_hits ) {
 
 			// only repair those subsystems which are not destroyed
-			if ( ssp->max_hits <= 0 || ssp->current_hits <= 0 )
+			if ( ssp->max_hits <= 0 || ssp->current_hits <= 0 && !(sip->flags2 & SIF2_SUBSYS_REPAIR_WHEN_DISABLED || ssp->flags & SSF_AUTOREPAIR_IF_DISABLED) )
```

Comment:
Oh right, we made warnings errors... but only for the Linux configuration, for some reason. Let me amend the commit to keep the history vaguely clean.